transport/src/main/java/io/netty/channel/nio/AbstractNioMessageChannel.java

```diff
@@ -91,6 +86,7 @@ public void read() {
 
                 int size = readBuf.size();
                 for (int i = 0; i < size; i ++) {
+                    setReadPending(false);
```

Comment:
should we better do an if check before enter the for loop and also only "update" the field one time.